sklearn/feature_extraction/dict_vectorizer.py

```diff
@@ -134,6 +131,32 @@ def fit(self, X, y=None):
 
         return self
 
+    def add_element(self, f, v, feature_names, vocab, fitting=True,
+                    transforming=False, indices=None, values=None):
+        if not isinstance(v, six.string_types) and isinstance(v, Iterable):
+            for vv in v:
```

Comment:
I don't think we want to have iterables of anything but strings. I think it would be worth our while raising an error here, even if that slows things down.